src/main.rs

```diff
@@ -133,6 +146,53 @@ async fn handle_request(req: Request<hyper::body::Incoming>, state: SharedState)
                     .boxed(),
             )?),
         (&Method::GET, "/list") => build_json_response(&*state.read().await),
+        (&Method::DELETE, "/delete") => {
+            let request: DeleteRequest =
+                serde_json::from_reader(req.collect().await?.aggregate().reader())?;
+            let mut state = state.write().await;
```

Comment:
Should be in block to minimize usage

Reply:
It should be fine if we remove from state first before deleting the file because can always add back